launcher/src/main/java/org/apache/spark/launcher/Main.java

```diff
@@ -47,10 +47,14 @@
    * character. On Windows, the output is a command line suitable for direct execution from the
    * script.
    */
+
+  static String uberJarPath;
```

Comment:
This looks really ugly. I'd really prefer plumbing this to the command builders through the constructor. It's a little bit more code but much cleaner.